Warehouse/CMakeLists.txt

```diff
@@ -31,6 +31,7 @@ set_target_properties(${MODULE_NAME} PROPERTIES
         CXX_STANDARD_REQUIRED YES)
 
 target_compile_definitions(${MODULE_NAME} PRIVATE MODULE_NAME=Plugin_${PLUGIN_NAME})
+target_compile_options(${MODULE_NAME} PRIVATE -Wno-error=attribute-warning)
```

Comment:
please remove changes to all CMakeLists.txt in this pull request. The task is to fix warnings, and not to hide them selectively

Reply:
@npoltorapavlo - Some of these need to be handled outside in the HAL before we can enable warnings as error. Srikanth is working with Amit to get that addressed. In the meantime, I would prefer to enable it asap on the recipe and have these exceptions for now. We will remove it once the HAL changes are done.

Reply:
@npoltorapavlo - I'll create different tickets to handle the suppressed errors and will remove these changes to CMakeLists.txt at that time.

Reply:
this broke the build on XG1V3
```
18:43:37 | cc1plus: error: -Werror=attribute-warning: no option -Wattribute-warning
```

Reply:
I am seeing the same thing on PACEXID, Ciscoxid also. I think they use different compiler versions that doesn't know about "attribute-warning" option.